### ISSUE TITLE
[keymgr] Reseed interval timer modfications

### ISSUE THREAD

**Issue by vogelpi** [lowRISC/opentitan#22877]
### Description

While debugging a ROM_EXT issue #22819, we've noticed that the reseed interval timer inside keymgr is active all the time and doesn't take into account whether keymgr is actually busy.

This means, once software has started keymgr, it will continuously request fresh entropy from EDN0 (by default 64 bits every 256 clock cycles). What's worse, if software decides to lock the [reseed interval register](https://opentitan.org/book/hw/ip/keymgr/doc/registers.html#reseed_interval_shadowed), there is no way to stop this afterwards. The max interval is 65536, meaning every 0.65 ms, a reseed is happening anyway. Because the reseed frequency is then not tied to activity, there is no way to program this adequately.

I am proposing to change the design as follows: The counter should not count the clock cycles but instead the number of PRNG updates. The PRNG is currently updated using `ctrl_lfsr_en | data_lfsr_en | sideload_lfsr_en`. I am suggesting to just feed this signal also into the reseed interval timer.

Any objections @moidx @johannheyszl @andreaskurth ?

**Comment by johannheyszl** [comment]
sgtm.
cc @ballifatih

**Comment by vogelpi** [comment]
Thanks for the feedback @johannheyszl and @ballifatih .

`reg2hw.start.q` was actually low in the waves I saw. keymgr de-asserts that after running an operation. So, using keymgr once and then leaving alone will trigger this behavior. I think we should investigate how to resolve / workaround the DV issue. A simple solution could be to not make the reseed counter count when start is disabled, i.e., after running a command also stop the counter.

**Comment by ballifatih** [comment]
I agree that it would be great if we can drop `~reg2hw.start.q` from `seed_en_i` signal. Then, we can use `reg2hw.start.q` to gate the counter.

This line of code was introduced very early on #4448, I had a look at the PR, but I could not figure out what it actually prevents on DV side.

**Comment by vogelpi** [comment]
Thanks for digging up that old PR - it seems I also reviewed that :-D 

I've now removed that `~reg2hw.start.q` qualifier for the seed enable and didn't get additional failures in block-level DV. I'll remove that as part of the PR. I think I'll try to do an implementation counting the number of LFSR updates as this is a more accurate measurement of how "old" the previous seed is. How, long KeyMgr operations take (as measured by  `~reg2hw.start.q`) can depend on other factors as well including the entropy complex (in case KMAC needs to perform a reseed operation as well).
